### ISSUE TITLE
Linked packages should have folder name which is used by dependant's package.json

### ISSUE THREAD

**Issue by KSXGitHub** [pnpm/pnpm#1364]
# TL;DR

When I write a different name (from dependency's own `package.json`) in `dependencies` field, I expect that name to be importable.

# Steps to reproduce

## 1. Create a monorepo structure

![Folder Structure](https://i.imgur.com/kuGEO35.png)

### 1.1. File `dependant/package.json`

```json
{
  "name": "dependant",
  "private": true,
  "version": "0.0.0",
  "dependencies": {
    "new-name": "file:../dependency"
  }
}
```

### 1.2. File `dependency/package.json`

```json
{
  "name": "old-name",
  "private": true,
  "version": "0.0.0"
}
```

## 2. Link them together

```sh
pnpm recursive link
```

# Expected behavior

pnpm creates a symlink at `dependant/node_modules/new-name`.

# Actual behavior

pnpm creates `dependant/node_modules/old-name` instead.

![PNPM](https://i.imgur.com/GZg4PKn.png)

# Notes

* Yarn not only creates `new-name` but also `old-name` and `dependant`, however, I don't think `old-name` and `dependant` should be created at all. [image](https://i.imgur.com/8Tb63wM.png)

**Comment by zkochan** [comment]
I think you can use an alias: `"new-name": "dependency@file:../dependency"`

**Comment by zkochan** [comment]
:ship: [2.15.2](https://github.com/pnpm/pnpm/releases/tag/v2.15.2)

**Comment by KSXGitHub** [comment]
Wait, why is pnpm is still 2.15.1 in npmjs.com?

[![image](https://i.imgur.com/DqGuQda.png)](https://www.npmjs.com/package/pnpm)

**Comment by zkochan** [comment]
yep, we don't make new versions `latest` immediately. If you need updates earlier, install with the `next` or `rc` tag
